### ISSUE TITLE
False positive type error with symbol argument + varargs

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#1978]
``` clojure
(defn foo
  ([^clojure.lang.Symbol var-sym]
   var-sym)
  ([^String a-ns
    ^String a-name
    & {:keys [debug?]}]       ; remove both this & keys ...
   (str a-ns a-name debug?))) ;         and `debug?`    ... and error goes away

(foo 'inc)
```

Workaround:

``` clojure
{:linters {:type-mismatch
           {:namespaces
            {dude
             {foo
              {:arities {1 {:args [:symbol]}
                         :varargs {:args [:string :string {:op :rest :spec :any}]}}}}}}}}
```
